Fix/remove uboot ota

lbonn opened the pull request:
u-boot support for minnowboard was not even building properly, let's remove it as it uses grub by default

lbonn committed:
Remove reference to non-existing yocto layer

lbonn committed:
Remove u-boot support for minnowboard

Was already broken

pattivacek commented on conf/bblayers.conf.sample:
```diff
@@ -11,7 +11,6 @@ BBLAYERS ?= " \
   ##OEROOT##/meta-yocto-bsp \
   ##OEROOT##/../meta-raspberrypi \
   ##OEROOT##/../meta-updater \
-  ##OEROOT##/../meta-updater-minnowmax \
```
This seems like it was a typo, I guess? It should be `meta-updater-minnowboard`. Otherwise by default our own layer won't be included...

lbonn replied:
Actually, this layer is included through this inclusion: https://github.com/advancedtelematic/meta-updater/blob/5f803ebaa590c408ef4b6e2a78e03e3af60161a2/conf/include/bblayers/sota_intel-corei7-64.inc. I am not exactly sure of why we have two different ways to do that (@OYTIS ?) but at least I am pretty sure that `meta-updater-minnowmax` is not relevant.

pattivacek replied:
I've looked into this before, and I forget what the `.sample` files are actually used for. While we don't have one for qemu, we do for meta-updater-raspberrypi, and it does include itself. So if we have this file, we should make it accurate, but if we don't need the file, then we should just remove it.

pattivacek commented:
Are the `recipes-bsp/bootfiles/` really unused or obsolete?

lbonn commented:
@patrickvacek it only ships `uEnv.txt` which is read by u-boot. I guess we should also remove this file then: https://github.com/advancedtelematic/meta-updater/blob/5f803ebaa590c408ef4b6e2a78e03e3af60161a2/classes/sota_minnowboard_uboot.inc

pattivacek commented:
@lbonn Yep, probably! I didn't realize we had all this u-boot cruft in here. I've only ever booted this with grub!

lbonn committed:
Cleanup erroneous layer templating

These are added later by `envsetup.sh` in meta-updater. Plus they
contradict our recommended directory structure.

lbonn commented:
@patrickvacek about the layer templating thing: after inspection, everything outside of poky doesn't belong there.

pattivacek commented:
> everything outside of poky doesn't belong there

I'm seeing other layers that show a similar list of layer-specific dependencies and such, i.e. `meta-agl/templates/base/bblayers.conf.sample`. I'm not sure it matters, though.

pattivacek commented:
@lbonn I confirm your assessment. It looks like non-meta-updater layers do use bblayers.conf.sample to construct a default, but we use `envsetup.sh` to do the job. I'm not sure which way is better, but for now, we should just be consistent with what we already do, so this looks good.
